cli/lib/wordpress-server-manager.ts

```diff
@@ -0,0 +1,252 @@
+ * - Has both inactivity timeout and max total timeout
+ */
+let nextMessageId = 0;
+
+async function sendMessage(
+	pmId: number,
+	type: string,
+	data: Message[ 'data' ]
+): Promise< unknown > {
+	const bus = await getPm2Bus();
+
+	return new Promise( ( resolve, reject ) => {
+		const id = nextMessageId++;
+		const message: Message = { id, type, data };
+
+		const startTime = Date.now();
+		let lastActivityTimestamp = Date.now();
+
+		const cleanup = () => {
+			bus.off( 'process:msg', responseHandler );
+			const tracker = activityTrackers.get( id );
+			if ( tracker ) {
+				clearInterval( tracker.activityCheckInterval );
+				activityTrackers.delete( id );
+			}
+		};
+
+		const activityCheckInterval = setInterval( () => {
+			const now = Date.now();
+			const timeSinceLastActivity = now - lastActivityTimestamp;
+			const totalElapsedTime = now - startTime;
+
+			if (
+				timeSinceLastActivity > PLAYGROUND_CLI_INACTIVITY_TIMEOUT ||
+				totalElapsedTime > PLAYGROUND_CLI_MAX_TIMEOUT
+			) {
+				cleanup();
+				const timeoutReason =
+					totalElapsedTime > PLAYGROUND_CLI_MAX_TIMEOUT
+						? `Maximum timeout of ${ PLAYGROUND_CLI_MAX_TIMEOUT / 1000 }s exceeded`
+						: `No activity for ${ PLAYGROUND_CLI_INACTIVITY_TIMEOUT / 1000 }s`;
+				reject(
+					new Error( `Timeout waiting for response to message ${ id }: ${ timeoutReason }` )
+				);
+			}
+		}, PLAYGROUND_CLI_ACTIVITY_CHECK_INTERVAL );
+
+		activityTrackers.set( id, {
+			lastActivityTimestamp,
+			activityCheckInterval,
+		} );
+
+		const responseHandler = ( packet: unknown ) => {
+			const validationResult = PacketSchema.safeParse( packet );
+			if ( ! validationResult.success ) {
+				return;
+			}
+
+			const validPacket = validationResult.data;
+
+			if ( validPacket.process.pm_id === pmId ) {
+				if (
+					validPacket.raw.type === 'activity' ||
+					( 'id' in validPacket.raw && validPacket.raw.id !== undefined )
+				) {
+					lastActivityTimestamp = Date.now();
+					const tracker = activityTrackers.get( id );
+					if ( tracker ) {
+						tracker.lastActivityTimestamp = lastActivityTimestamp;
+					}
+				}
+			}
+
+			if (
+				validPacket.process.pm_id === pmId &&
+				'id' in validPacket.raw &&
+				validPacket.raw.id === id
+			) {
+				cleanup();
+
+				if ( 'error' in validPacket.raw && validPacket.raw.error ) {
+					const error = new Error( validPacket.raw.error ) as Error & {
+						errorStack?: string;
+					};
+					if ( 'errorStack' in validPacket.raw && validPacket.raw.errorStack ) {
+						error.stack = validPacket.raw.errorStack;
+					}
+					reject( error );
+				} else {
+					resolve( 'result' in validPacket.raw ? validPacket.raw.result : undefined );
+				}
+			}
+		};
+
+		bus.on( 'process:msg', responseHandler );
+
+		pm2.sendDataToProcessId(
+			pmId,
+			{
+				type: 'process:msg',
+				data: message,
+				topic: true,
+			},
+			( error ) => {
+				if ( error ) {
+					cleanup();
+					reject( error );
+				}
+			}
+		);
```

Comment:
```suggestion
		pm2.sendDataToProcessId( pmId, message, ( error ) => {
			if ( error ) {
				cleanup();
				reject( error );
			}
		} );
```
Let's use the `topic` property. pm2 requires this prop to be defined.